WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift

```diff
@@ -791,3 +729,84 @@ private extension CommentDetailViewController {
     }
 
 }
+
+// MARK: - UITableView Methods
+
+extension CommentDetailViewController: UITableViewDelegate, UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return rows.count
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = rows[indexPath.row]
+        let cell: UITableViewCell = {
+            switch row {
+            case .header:
+                configureHeaderCell()
+                return headerCell
+
+            case .content:
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: CommentContentTableViewCell.defaultReuseID) as? CommentContentTableViewCell else {
+                    return .init()
+                }
+
+                configureContentCell(cell, comment: comment)
+                cell.moderationBar.delegate = self
+                moderationBar = cell.moderationBar
+                return cell
+
+            case .replyIndicator:
+                return replyIndicatorCell
+
+            case .text:
+                return configuredTextCell(for: row)
+
+            case .deleteComment:
+                return deleteButtonCell
+            }
+        }()
+
+        // hide cell separator if it's positioned before the delete button cell.
+        cell.separatorInset = shouldHideCellSeparator(for: indexPath) ? insetsForHiddenCellSeparator : tableView.separatorInset
+
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row] {
+        case .header:
+            comment.hasParentComment() ? navigateToParentComment() : navigateToPost()
+
+        case .replyIndicator:
+            // TODO: Navigate to the comment reply.
+            break
```

Comment:
```suggestion
            navigateToReplyComment()
```

I'm not sure why this is reverted 🤔 , but since this PR is based on #17409, this should navigate to the reply comment properly.

Reply:
Well you're assuming I didn't do something dumb. Which clearly I did. 😆  Thank you!